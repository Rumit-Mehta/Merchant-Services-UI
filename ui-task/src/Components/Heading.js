import React from "react";

const Heading = () => {
  return (
    <div className="headingDiv">
      <h1 id="heading">Assign Services</h1>
      <h2 id="subheading">Set up Client</h2>
    </div>
  );
};

export default Heading;
