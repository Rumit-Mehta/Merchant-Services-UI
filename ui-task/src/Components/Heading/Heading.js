import React from "react";

const Heading = () => {
  return (
    <div className="headingDiv">
      <div className="headingAccent"></div>
      <div className="headingText">
        <h1 data-testid="headingId" id="heading">
          Assign services
        </h1>
        <h2 data-testid="subheadingId" id="subheading">
          Set up client
        </h2>
      </div>
    </div>
  );
};

export default Heading;
