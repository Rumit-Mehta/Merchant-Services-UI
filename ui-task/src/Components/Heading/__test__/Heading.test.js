import React from "react";
import ReactDOM from "react-dom";
import Heading from "../Heading";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

test("renders  without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Heading></Heading>, div);
});

test("renderes Heading correctly", () => {
  const { getByTestId } = render(<Heading />);
  expect(getByTestId("headingId")).toHaveTextContent("Assign services");
});

test("renderes subheading correctly", () => {
  const { getByTestId } = render(<Heading />);
  expect(getByTestId("subheadingId")).toHaveTextContent("Set up client");
});

test("matches snapshot", () => {
  const tree = renderer.create(<Heading />).toJSON();
  expect(tree).toMatchSnapshot();
});
