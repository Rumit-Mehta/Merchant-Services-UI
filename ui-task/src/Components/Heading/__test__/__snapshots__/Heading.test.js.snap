// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="headingDiv"
>
  <div
    className="headingAccent"
  />
  <div
    className="headingText"
  >
    <h1
      data-testid="headingId"
      id="heading"
    >
      Assign services
    </h1>
    <h2
      data-testid="subheadingId"
      id="subheading"
    >
      Set up client
    </h2>
  </div>
</div>
`;
